### PR TITLE
Backport 6942632: Hotspot should be able to use more than 64 logical processors on Windows

### PR DESCRIPTION
**Pull request by swesonga**
Merge conflicts resolved in these files:

make/test/JtregNativeHotspot.gmk
src/hotspot/os/windows/globals_windows.hpp
src/hotspot/os/windows/os_windows.cpp
test/hotspot/gtest/runtime/test_os_windows.cpp
test/hotspot/jtreg/TEST.groups

### CODE REVIEW

File: ms-patches/use-all-windows-processor-groups-6942632.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+title: Hotspot should be able to use more than 64 logical processors on Windows
+- work_item: 2114223
+- jbs_bug: JDK-6942632
+- author: swesonga
+- owner: swesonga
+- contributors:
+- details:
+  - Backport of https://github.com/openjdk/jdk/pull/17576.
+- release_note:
+  - Enables hotspot to use processors across all Windows processor groups on Windows 11/Windows Server 2022 and later.
```
Comment (brianjstafford) [inline ms-patches/use-all-windows-processor-groups-6942632.yml:10]: nit: Hotspot

File: test/hotspot/gtest/runtime/test_os_windows.cpp [hunk 2]
```diff
@@ -617,4 +617,23 @@ TEST_VM(os_windows, handle_long_paths) {
   delete_empty_rel_directory_w(nearly_long_rel_path);
 }
 
+TEST_VM(os_windows, processor_count) {
```
Comment (brianjstafford) [inline test/hotspot/gtest/runtime/test_os_windows.cpp:620]: What was the merge conflict in this file?

**Comment by brianjstafford** [comment]
LGTM

File: test/hotspot/gtest/runtime/test_os_windows.cpp (hunk 2 above)
Comment (swesonga) [reply test/hotspot/gtest/runtime/test_os_windows.cpp:620]: > What was/were the merge conflict(s) in this file?

My change appended the processor_count test to this file and another change appended the reserve_memory_special_concurrent test to the same file. I edited the merge to select my test only.

**Pull request merged**